ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -2960,8 +3004,18 @@ static void opt_for_reorder(ggml_backend_sycl_context * ctx, const ggml_tensor *
     extra->optimized_feature.reorder = true;  // Used to decode/dequan in next steps and avoid re-reordering
 }
 
-static void ggml_sycl_mul_mat(ggml_backend_sycl_context & ctx, const ggml_tensor * src0, const ggml_tensor * src1, ggml_tensor * dst) {
 
+static bool can_use_dequantize_mul_mat_vec(const ggml_tensor * src0, const ggml_tensor * src1, ggml_tensor * dst) {
```

Comment:
```suggestion
static bool choose_dequantize_mul_mat_vec(const ggml_tensor * src0, const ggml_tensor * src1, ggml_tensor * dst) {
```

Reply:
It's nit but `can_use` seems more accurate to me since there are more logic later on to make the final decision on the matmul implementation.

Reply:
We should use one function to detect if need call deq_mul_mat_vec(), instead of several.

Reply:
What I mean is that the final choice depends on the outputs from `can_use_dequantize_mul_mat_vec` and `can_use_mul_mat_vec_q` so it can't all be in a single `choose_dequantize_mul_mat_vec` currently.

Reply:
Yes, I agree with @Rbiessy and that's why it is currently implemented this way.